notebooks/utils/utils.py

```diff
@@ -100,3 +100,46 @@ def get_varnames_from_metadata_list(diag_metadata_list):
         if diag_metadata["varname"] not in varnames:
             varnames.append(diag_metadata["varname"])
     return varnames
+
+
+def timeseries_and_history_comparison(casename):
+    # import here to avoid "ValueError: attempted relative import beyond top-level package"
+    # when Github Action calls "import utils" (no issue in notebooks)
+    from .compare_ts_and_hist import compare_ts_and_hist
```

Comment:
@andersy005 can you suggest a better approach to this? If I put the import statement at the top of the file, then Github Actions fail. In that situation I can recreate the error with

```
$ cd notebooks
$ python
>>> import utils
```

There's a potential for some circular dependencies which might be at play? `compare_ts_and_hist` needs `CaseClass` out of `utils/` so I think including it in `__init__.py` causes problems. (Also, I'm confused about why jupyter doesn't complain about this.)

Reply:
@mnlevy1981, try the following 


```diff
$ git --no-pager diff notebooks/utils/
diff --git a/notebooks/utils/utils.py b/notebooks/utils/utils.py
index ec576d1..24e5322 100644
--- a/notebooks/utils/utils.py
+++ b/notebooks/utils/utils.py
@@ -5,6 +5,7 @@ import math
 import cftime
 import numpy as np
 import xarray as xr
+from .compare_ts_and_hist import compare_ts_and_hist
 
 
 def repl_coord(coordname, ds1, ds2):
@@ -105,7 +106,7 @@ def get_varnames_from_metadata_list(diag_metadata_list):
 def timeseries_and_history_comparison(casename):
     # import here to avoid "ValueError: attempted relative import beyond top-level package"
     # when Github Action calls "import utils" (no issue in notebooks)
-    from .compare_ts_and_hist import compare_ts_and_hist
+    
 
     for year in range(1, 62):
         has_ts = True
```


```diff
$ git --no-pager diff tests/
diff --git a/tests/test_utils.py b/tests/test_utils.py
index ccaf396..7d025d1 100755
--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -7,9 +7,10 @@ import cftime
 import numpy as np
 import xarray as xr
 
-sys.path.append(os.path.abspath(os.path.join("notebooks", "utils")))
+
+sys.path.append(os.path.abspath("notebooks"))
 sys.path.append(os.path.abspath("tests"))
-from utils import time_year_plus_frac, time_set_mid, repl_coord, round_sig
+from utils.utils import time_year_plus_frac, time_set_mid, repl_coord, round_sig
 from xr_ds_ex import gen_time_bounds_values, xr_ds_ex
 
 nyrs = 300
```

Reply:
This fixed it, thanks! It makes sense to update something in the tests when the tests failed but the notebooks worked, but I didn't even think to look in there.